src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexMatch.cs

```diff
@@ -433,11 +433,9 @@ internal override void Dump()
         {
             if (_caps != null)
             {
-                IEnumerator<Int32> e = _caps.Keys.GetEnumerator();
-
-                while (e.MoveNext())
+                foreach (KeyValuePair<int, int> kvp in _caps)
```

Comment:
:bulb: Any reason to not wrap this in `#if DEBUG` to avoid the iteration? Or perhaps move the block to a `[Conditional("DEBUG")]` method?